src/java.base/share/classes/java/text/ChoiceFormat.java

```diff
@@ -541,8 +554,18 @@ public StringBuffer format(double number, StringBuffer toAppendTo,
         }
         --i;
         if (i < 0) i = 0;
-        // return either a formatted number, or a string
-        return toAppendTo.append(choiceFormats[i]);
+        try {
+            // return either a formatted number, or a string
+            toAppendTo.append(choiceFormats[i]);
+        } catch (IOException ioe) {
+            throw new UncheckedIOException(ioe.getMessage(), ioe);
```

Comment:
Perhaps `AssertionError` instead of `UncheckedIOException` is better suited here and in the other ocurrences.

Reply:
This can be removed if using `StringBuf`.